hw/chisel/src/main/scala/snax/streamer/DesignParams.scala

```diff
@@ -69,6 +73,30 @@ class StreamerParam(
     )
   }
 
+  // transpose parameters
+  if (hasTranspose) {
+    require(
+      readerNum == 2,
+      "Transpose only supports for gemm with 2 readers"
+    )
+    require(
+      fifoWidthReader(0) == fifoWidthReader(1) && fifoWidthReader(0) == 512,
+      "Transpose only supports for gemm with 2 readers with the same 512 data width"
+    )
```

Comment:
this should be a function of tcdmParam.dataWidth, no?

Reply:
No, this is fixed to 512 otherwise the transpose extension won't work.